### PR TITLE
Adding db2dframes function

### PR DESCRIPTION
**Pull request by afard**
Shreya (@shreya2k7),

Can you please take care of code review for the new db2dframes function?

Thanks,
Arash

### CODE REVIEW

File: algorithms/HPdata/man/db2dframes.Rd [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    the list of the column names corresponding to responses.
+  }
+  \item{pred}{
+    this is an optional argument to specify list of the column names corresponding to predictors. If this argument is not specfied or is empty, the function will load all columns of the table or view excluding the column specified in resp argument.
+  }
+  \item{except}{
+    this is an optional argument to specify the list of column names that should be excluded from pred.
+  }
+  \item{npartitions}{
+    this optional argument specifies the desired number of partitions in the distributed array. If not specified, it will be equal to the number of active R instances in the distributed R cluster. It should be noted that the number of splits in the returned result might not be exactly the same as npartitions. Please read the details for more information.
```
Comment (shreya2k7) [inline algorithms/HPdata/man/db2dframes.Rd]: array should be replaced with data.frame

File: algorithms/HPdata/man/db2dframes.Rd [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  \item{loadPolicy}{
+    it is the data loading policy used for Vertica Connector for Distributed R. It can have two values, "local" or "uniform". By default, "local" policy is used. Views and external tables will always be loaded with "uniform" policy. See details section for more information. 
+  }
+}
+
+\details{
+The number of partitions of the resultant dframes will be approximately equal to value provided in argument 'npartitions'.
+When Vertica Connector for Distributed R is used, argument 'loadPolicy' defines the data loading policy between Distributed R and Vertica. loadPolicy "local" is employed when number of Distributed R worker is same as number for Vertica nodes containing data for table specified in argument 'tableName'. With this policy, there is one-to-one data load from a Vertica node to a Distributed R worker. For the alternate scenario, loadPolicy "uniform" is used where each Vertica nodes involves all Distributed R nodes in the data loading process. By default, loadPolicy "local" is used but it is automatically changed to "uniform" when the conditions for loadPolicy "local" are not met.
+
+db2dframes supports numeric and logical data types.
```
Comment (shreya2k7) [inline algorithms/HPdata/man/db2dframes.Rd]: it supports other data types like characters also

File: algorithms/HPdata/R/db2dframes.R [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+          dResult <- .vertica.dframes(result, nResponses, nPredictors, pred_columns, resp)
+
+          }, interrupt = function(e) {}
+           , error = function(e) {
+             .vertica.connector(e)
+          } , finally = { 
+             stopDataLoader()
+          })
+
+    # end of verticaConnector
```
Comment (shreya2k7) [inline algorithms/HPdata/R/db2dframes.R]: I guess for NDL, the code is same as db2darrays() except the call to .vertica,dframes()

**Comment by shreya2k7** [comment]
Completed code review. Looks good overall.

File: algorithms/HPdata/R/db2dframes.R (hunk 3 above)
Comment (afard) [inline algorithms/HPdata/R/db2dframes.R]: That is almost correct. I have also changed the column type validation to be similar to db2dframe.

**Comment by afard** [comment]
Thanks Shreya! I make your suggested corrections in the help page.

**Pull request merged**